### DISCUSSION TITLE
Can the spot light support transformManager.setParent

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#4568] [Q&A] [answered]
I use the API transformManager.setParent, It can work fine for a mesh object.
But for the spot light like below code:
tm.setParent(tm.getInstance(spotLight),tm.getInstance(root))
It can't work.

**Comment by romainguy** [comment]
It should work just fine. Lights aren't treated differently.

**Comment by fanbinnet** [comment, marked as answer]
I found  my mistake,light don't have default transform,shoud first create tm for it.
tm.create(spotLight2,tm.getInstance(root),transformMatrix)
